src/TriggerBinding/SqlTriggerListener.cs

```diff
@@ -0,0 +1,364 @@
+            {
+                using (SqlDataReader reader = await getUserTableColumnsCommand.ExecuteReaderAsync(cancellationToken))
+                {
+
+                    var userTableColumns = new List<string>();
+
+                    while (await reader.ReadAsync(cancellationToken))
+                    {
+                        userTableColumns.Add(reader.GetString(0));
+                    }
+
+                    return userTableColumns;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Creates the schema where the worker tables will be located if it does not already exist.
+        /// </summary>
+        private static async Task CreateSchemaAsync(SqlConnection connection, SqlTransaction transaction, CancellationToken cancellationToken)
+        {
+            string createSchemaQuery = $@"
+                IF SCHEMA_ID(N'{SqlTriggerConstants.SchemaName}') IS NULL
+                    EXEC ('CREATE SCHEMA [{SqlTriggerConstants.SchemaName}]');
+            ";
+
+            using (var createSchemaCommand = new SqlCommand(createSchemaQuery, connection, transaction))
+            {
+                await createSchemaCommand.ExecuteNonQueryAsync(cancellationToken);
+            }
+        }
+
+        /// <summary>
+        /// Creates the global state table if it does not already exist.
+        /// </summary>
+        private static async Task CreateGlobalStateTableAsync(SqlConnection connection, SqlTransaction transaction, CancellationToken cancellationToken)
+        {
+            string createGlobalStateTableQuery = $@"
+                IF OBJECT_ID(N'{SqlTriggerConstants.GlobalStateTableName}', 'U') IS NULL
+                    CREATE TABLE {SqlTriggerConstants.GlobalStateTableName} (
+                        UserFunctionID char(16) NOT NULL,
+                        UserTableID int NOT NULL,
+                        LastSyncVersion bigint NOT NULL,
+                        PRIMARY KEY (UserFunctionID, UserTableID)
+                    );
+            ";
+
+            using (var createGlobalStateTableCommand = new SqlCommand(createGlobalStateTableQuery, connection, transaction))
+            {
+                await createGlobalStateTableCommand.ExecuteNonQueryAsync(cancellationToken);
+            }
+        }
+
+        /// <summary>
+        /// Inserts row for the user table and the function inside the global state table, if one does not already exist.
+        /// </summary>
+        private async Task InsertGlobalStateTableRowAsync(SqlConnection connection, SqlTransaction transaction, int userTableId, CancellationToken cancellationToken)
+        {
+            string insertRowGlobalStateTableQuery = $@"
+                IF NOT EXISTS (
+                    SELECT * FROM {SqlTriggerConstants.GlobalStateTableName}
+                    WHERE UserFunctionID = '{this._userFunctionId}' AND UserTableID = {userTableId}
```

Comment:
Is this function ID guaranteed not to have single quotes in it? (you should probably escape it regardless just to be safe)

Reply:
I do think so, as it's a hash, but have escaped it regardless during initialization of function ID.